src/biome/text/api_new/modules/heads/bimpm_classification.py

```diff
@@ -0,0 +1,487 @@
+        """
+        record1_field = self.model.featurize(record1, aggregate=not self.multifield)
+        record2_field = self.model.featurize(record2, aggregate=not self.multifield)
+        instance = Instance(
+            {
+                "record1": record1_field.get("record"),
+                "record2": record2_field.get("record"),
+            }
+        )
+        return self.add_label(instance, label)
+
+    @overrides
+    def forward(
+        self,  # type: ignore
+        record1: Dict[str, torch.LongTensor],
+        record2: Dict[str, torch.LongTensor],
+        label: torch.LongTensor = None,
+    ) -> Dict[str, torch.Tensor]:
+        # pylint: disable=arguments-differ
+        """
+        Parameters
+        ----------
+        record1
+            The first input tokens.
+            The dictionary is the output of a ``*Field.as_array()``. It gives names to the tensors created by
+            the ``TokenIndexer``s.
+            In its most basic form, using a ``SingleIdTokenIndexer``, the dictionary is composed of:
+            ``{"tokens": Tensor(batch_size, num_tokens)}``.
+            The keys of the dictionary are defined in the `pipeline.yaml` config.
+            The dictionary is designed to be passed on directly to a ``TextFieldEmbedder``, that has a
+            ``TokenEmbedder`` for each key in the dictionary (except you set `allow_unmatched_keys` in the
+            ``TextFieldEmbedder`` to False) and knows how to combine different word/character representations into a
+            single vector per token in your input.
+        record2
+            The second input tokens.
+        label : torch.LongTensor, optional (default = None)
+            A torch tensor representing the sequence of integer gold class label of shape
+            ``(batch_size, num_classes)``.
+
+        Returns
+        -------
+        An output dictionary consisting of:
+        logits : torch.FloatTensor
+            A tensor of shape ``(batch_size, num_tokens, tag_vocab_size)`` representing
+            unnormalised log probabilities of the tag classes.
+        class_probabilities : torch.FloatTensor
+            A tensor of shape ``(batch_size, num_tokens, tag_vocab_size)`` representing
+            a distribution of the tag classes per word.
+        loss : torch.FloatTensor, optional
+            A scalar loss to be optimised.
+        """
+        mask_record1 = util.get_text_field_mask(
+            record1, num_wrapping_dims=self.num_wrapping_dims
+        )
+        mask_record2 = util.get_text_field_mask(
+            record2, num_wrapping_dims=self.num_wrapping_dims
+        )
+
+        # embedding and encoding of record1
+        embedded_record1 = self.dropout(
+            self.model.forward(
+                record1, mask=mask_record1, num_wrapping_dims=self.num_wrapping_dims
+            )
+        )
+        encoded_record1 = self.dropout(
+            self.td_encoder(embedded_record1, mask=mask_record1)
```

Comment:
When moving this to backbone model we can do something like `self.model.encoder`

Reply:
I used the new `TimeDistributedEncoder` to remove the need for the separate `*.td_encoder`s.